argoverse/evaluation/stereo/eval.py

```diff
@@ -0,0 +1,348 @@
+
+        all:10,
+        fg:10,
+        bg:10,
+        all*:10,
+        fg*:10,
+        bg*:10,
+        all:5,
+        fg:5,
+        bg:5,
+        all*:5,
+        fg*:5,
+        bg*:5,
+        all:3,
+        fg:3,
+        bg:3,
+        all*:3,
+        fg*:3,
+        bg*:3.
+
+    Note: The `evaluate` function will use all available logical cores on the machine.
+
+"""
+import argparse
+import logging
+import os
+from multiprocessing import Pool
+from pathlib import Path
+from typing import DefaultDict, List
+
+import cv2
+import disparity_interpolation
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+log_colormap = [
+    [np.array([0, 0.0625]), np.array([49, 54, 149], dtype=np.uint8)],
+    [np.array([0.0625, 0.125]), np.array([69, 117, 180], dtype=np.uint8)],
+    [np.array([0.125, 0.25]), np.array([116, 173, 209], dtype=np.uint8)],
+    [np.array([0.25, 0.5]), np.array([171, 217, 233], dtype=np.uint8)],
+    [np.array([0.5, 1]), np.array([224, 243, 248], dtype=np.uint8)],
+    [np.array([1, 2]), np.array([254, 224, 144], dtype=np.uint8)],
+    [np.array([2, 4]), np.array([253, 174, 97], dtype=np.uint8)],
+    [np.array([4, 8]), np.array([244, 109, 67], dtype=np.uint8)],
+    [np.array([8, 16]), np.array([215, 48, 39], dtype=np.uint8)],
+    [np.array([16, np.Inf]), np.array([165, 0, 38], dtype=np.uint8)],
+]
+
+
+class StereoEvaluator:
+    """Instantiates a StereoEvaluator object for evaluation."""
+
+    def __init__(
+        self,
+        pred_root_fpath: Path,
+        gt_root_fpath: Path,
+        figs_fpath: Path,
+        abs_error_thresholds: List = [10, 5, 3],
+        rel_error_thresholds: List = [0.1, 0.1, 0.1],
+        save_disparity_error_image: bool = False,
+        num_procs: int = -1,
+    ) -> None:
+        """
+        Args:
+            pred_fpath_root: Path to the folder which contains the stereo predictions.
+            gt_fpath_root: Path to the folder which contains the stereo ground truth.
+            figs_fpath: Path to the folder which will contain the output figures.
+            num_procs: Number of processes among which to subdivide work.
+                Specifying -1 will use one process per available core
+        """
+        self.pred_root_fpath = pred_root_fpath
+        self.gt_root_fpath = gt_root_fpath
+        self.figs_fpath = figs_fpath
+        self.abs_error_thresholds = abs_error_thresholds
+        self.rel_error_thresholds = rel_error_thresholds
+        self.save_disparity_error_image = save_disparity_error_image
+        self.num_procs = os.cpu_count() if num_procs == -1 else num_procs
+
+    def evaluate(self) -> pd.DataFrame:
+        """Evaluate stereo output and return metrics. The multiprocessing
+        library is used for parallel processing of disparities.
```

Comment:
Maybe we could clarify what `parallel processing of disparities` means -- `parallel processing of disparity evaluation`?

Reply:
Changed to "...parallel processing of disparity evaluation".

Reply:
Agreed with Ben, I think it's worth considering a meta class for the evaluation scripts.